src/PreCICE.jl

```diff
@@ -618,18 +686,82 @@ Set mesh Quad from vertex IDs.
 - `vertices::AbstractArray{Integer}`: IDs of the edges of the Quads.
 
 """
-function setMeshQuad(meshName::String, vertices::AbstractArray{Integer})
+function setMeshQuads(meshName::String, vertices::AbstractArray{Integer})
+    _size, dimensions = size(vertices)
+    @assert dimensions == 4 "Dimensions of vector data in write_vector_data does not match with dimensions in problem definition. Provided dimensions: $dimensions, expected dimensions: $(getDimensions())"
     ccall(
         (:precicec_setMeshQuads, "libprecice"),
         Cvoid,
         (Ptr{Int8}, Cint, Ref{Cint}),
         meshName,
-        length(vertices),
-        vertices,
+        _size,
+        reshape(vertices, :),
     )
 
 end
 
+@doc """
+
+    setMeshTetrahedron(meshName::String, firstEdgeID::Integer, secondEdgeID::Integer, thirdEdgeID::Integer, fourthEdgeID::Integer)
+
+Set mesh Tetrahedron from edge IDs.
+
+# Arguments
+- `meshName::String`: Name of the mesh to add the Tetrahedron to.
+- `firstEdgeID::Integer`: ID of the first edge of the Tetrahedron.
+- `secondEdgeID::Integer`: ID of the second edge of the Tetrahedron.
+- `thirdEdgeID::Integer`: ID of the third edge of the Tetrahedron.
+- `fourthEdgeID::Integer`: ID of the fourth edge of the Tetrahedron.
+
+# Notes
+
+Previous calls:
+ - Edges with `first_edge_id`, `second_edge_id`, `third_edge_id`, and `fourth_edge_id` were added
+    to the mesh with the ID `mesh_id`
+"""
+function setMeshTetrahedron(
+    meshName::String,
+    firstEdgeID::Integer,
+    secondEdgeID::Integer,
+    thirdEdgeID::Integer,
+    fourthEdgeID::Integer,
+)
+    ccall(
+        (:precicec_setMeshTetrahedron, "libprecice"),
+        Cvoid,
+        (Ptr{Int8}, Cint, Cint, Cint, Cint),
+        meshName,
+        firstEdgeID,
+        secondEdgeID,
+        thirdEdgeID,
+        fourthEdgeID,
+    )
+end
+
+@doc """
+
+    setMeshTetrahedra(meshName::String, vertices::AbstractArray{Integer})
+
+Set mesh Tetrahedron from vertex IDs.
+
+# Arguments
+- `meshName::String`: Name of the mesh to add the Tetrahedron to.
+- `vertices::AbstractArray{Integer}`: IDs of the edges of the Tetrahedra.
+
+"""
+function setMeshTetrahedra(meshName::String, vertices::AbstractArray{Integer})
+    _size, dimensions = size(vertices)
+    @assert dimensions == 4 "Dimensions of vector data in write_vector_data does not match with dimensions in problem definition. Provided dimensions: $dimensions, expected dimensions: $(getDimensions())"
```

Comment:
Same as above. Also lets find a better name for the variable than **dimensions**.